File: apps/web/types/inspiration.ts
export type NotionStatus = "New" | "Processing" | "Processed" | "Failed";
export type Availability = "full_media" | "transcript_only" | "metadata_only" | "unavailable" | null;

export interface InspirationItemOut {
  id: string;
  notion_page_id: string;
  title: string | null;
  tiktok_url: string;
  tiktok_video_id: string | null;
  creator_name: string | null;
  topics: string[] | null;
  format_hint: string | null;
  note_why_saved: string | null;
  note_favorite_part: string | null;
  notion_status: NotionStatus;
  availability: Availability;
  thumbnail_url: string | null;
  embed_html: string | null;
  processed_at: string | null;
  error_message: string | null;
  already_used: boolean;
  created_at: string;
}

export interface SyncRunOut {
  id: string;
  source: string;
  status: string;
  started_at: string | null;
  finished_at: string | null;
  items_processed: number;
  items_failed: number;
  error_summary: string | null;
}

File: apps/api/app/schemas/inspiration.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class InspirationItemOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    notion_page_id: str
    title: str | None
    tiktok_url: str
    tiktok_video_id: str | None
    creator_name: str | None
    topics: list[str] | None
    format_hint: str | None
    note_why_saved: str | None
    note_favorite_part: str | None
    notion_status: str
    availability: str | None
    thumbnail_url: str | None
    embed_html: str | None
    processed_at: datetime | None
    error_message: str | None
    already_used: bool
    created_at: datetime


class SyncRunOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    source: str
    status: str
    started_at: datetime | None
    finished_at: datetime | None
    items_processed: int
    items_failed: int
    error_summary: str | None
